android/app/src/main/java/net/minetest/minetest/MainActivity.java

```diff
@@ -83,13 +84,18 @@ public void onReceive(Context context, Intent intent) {
 		}
 	};
 
+	@SuppressLint("UnspecifiedRegisterReceiverFlag")
 	@Override
 	public void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 		setContentView(R.layout.activity_main);
 
 		IntentFilter filter = new IntentFilter(ACTION_UPDATE);
-		registerReceiver(myReceiver, filter);
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+			registerReceiver(myReceiver, filter, RECEIVER_EXPORTED);
```

Comment:
This needs to be exported to allow the service to call it. Perhaps there's a flag that allows the service to call this when NOT_EXPORTED (maybe FLAG_IMMUTABLE)

Reply:
for communication inside an app the android docs recommend this instead: https://developer.android.com/develop/background-work/services/bound-services

Reply:
IntentService is deprecated so if we're switching to newer APIs then it would be worth starting with that